torchvision/datasets/celeba.py

```diff
@@ -146,10 +152,15 @@ def download(self) -> None:
             print("Files already downloaded and verified")
             return
 
-        for (file_id, md5, filename) in self.file_list:
-            download_file_from_google_drive(file_id, os.path.join(self.root, self.base_folder), filename, md5)
+        raise ValueError(
+            "Downloading CelebA is not supported anymore as of 0.13. See "
+            "https://github.com/pytorch/vision/issues/5705 for more details."
+        )
+        # TODO maybe uncomment lines below to put back download if it ever gets fixed.
+        # for (file_id, md5, filename) in self.file_list:
+        #     download_file_from_google_drive(file_id, os.path.join(self.root, self.base_folder), filename, md5)
 
-        extract_archive(os.path.join(self.root, self.base_folder, "img_align_celeba.zip"))
+        # extract_archive(os.path.join(self.root, self.base_folder, "img_align_celeba.zip"))
```

Comment:
Should we keep this? Otherwise the user does need to download *and* extract.

Reply:
Not there, but we could adapt the integrity check to check if we find only the archive and extract it:

https://github.com/pytorch/vision/blob/77cad12779a237c6fb415f48243bc149e9ecc36a/torchvision/datasets/celeba.py#L141-L142

Reply:
Do we extract on behalf of the user on other datasets where download isn't available?
I won't have time to do that today unfortunately, considering we have to test all this manually

Reply:
IIRC, `ImageNet` is the only dataset where we deactivated the download after it was available at first. For it, we do extract for the users:

https://github.com/pytorch/vision/blob/77cad12779a237c6fb415f48243bc149e9ecc36a/torchvision/datasets/imagenet.py#L61-L65

Other datasets that don't have a download functionality at all assume that the users do it themselves

https://github.com/pytorch/vision/blob/d57f929da98f927365bdef303d3453712067e15b/torchvision/datasets/coco.py#L15-L16

(Don't mind the "download" in the description. That is wrong as we don't offer anything for COCO)

Reply:
Thanks for the details.
Since in the majority of cases we don't extract for on behalf of users, I'd prefer merging this and just add to the instructions that they need to extract themselves (I updated the message in a previous commit)